src/utils/embedding_store.py

```
@@ -0,0 +1,425 @@
+
+        Args:
+            explicit: Number of explicit observations to retrieve
+            deductive: Number of deductive observations to retrieve
+        """
+        if explicit is not None:
+            self.explicit_observations_count = explicit
+        if deductive is not None:
+            self.deductive_observations_count = deductive
+
+    @observe()
+    async def save_unified_observations(
```

Ruff (0.12.2)

`save_unified_observations` is too complex (14 > 10)

(C901)